src/gs-background-tile.c

```diff
@@ -0,0 +1,276 @@
+
+	gtk_widget_show (tile->price);
+	gtk_widget_hide (tile->installed_icon);
+}
+
+static gboolean
+app_state_changed_idle (gpointer user_data)
+{
+	GsBackgroundTile *tile = GS_BACKGROUND_TILE (user_data);
+	g_autofree gchar *name = NULL;
+	AtkObject *accessible = gtk_widget_get_accessible (GTK_WIDGET (tile));
+
+	if (gs_app_is_installed (tile->app))
+		name = g_strdup_printf (_("%s (Installed)"), gs_app_get_name (tile->app));
+	else if (gs_app_get_state (tile->app) == AS_APP_STATE_AVAILABLE)
+		name = g_strdup (gs_app_get_name (tile->app));
+
+	if (GTK_IS_ACCESSIBLE (accessible)) {
+		atk_object_set_name (accessible, name);
+		atk_object_set_description (accessible, gs_app_get_summary (tile->app));
+	}
+
+	if (gs_app_get_metadata_item (tile->app, "GnomeSoftware::BackgroundTile-css") == NULL)
+		gs_background_tile_setup_colors_bg (tile);
+
+	gs_background_tile_update_price (tile);
+
+	g_object_unref (tile);
+	return G_SOURCE_REMOVE;
+}
+
+static void
+app_state_changed (GsApp *app, GParamSpec *pspec, GsBackgroundTile *tile)
+{
+	/* this is because the state-change notification could be
+	 * coming from a different thread */
+	g_idle_add (app_state_changed_idle, g_object_ref (tile));
+}
+
+static void
+app_background_tile_css_added (GsApp *app, const char *metadata, GsBackgroundTile *tile)
+{
+	const gchar *css = gs_app_get_metadata_item (app, "GnomeSoftware::BackgroundTile-css");
+	/* gs_background_tile_setup_colors_bg (tile); */
+	gs_utils_widget_set_css (GTK_WIDGET (tile->image_box), css);
+}
+
+
+static void
+gs_background_tile_set_app (GsAppTile *app_tile, GsApp *app)
+{
+	GsBackgroundTile *tile = GS_BACKGROUND_TILE (app_tile);
+	const gchar *summary = NULL;
+	const gchar *css;
+
+	g_return_if_fail (GS_IS_APP (app) || app == NULL);
+
+	if (tile->app)
+		g_signal_handlers_disconnect_by_func (tile->app, app_state_changed, tile);
+
+	g_set_object (&tile->app, app);
+	if (app == NULL)
+		return;
+
+	gtk_stack_set_visible_child_name (GTK_STACK (tile->stack), "content");
+
+	g_signal_connect (tile->app, "notify::state",
+			  G_CALLBACK (app_state_changed), tile);
+	g_signal_connect (tile->app, "notify::name",
+			  G_CALLBACK (app_state_changed), tile);
+	g_signal_connect (tile->app, "notify::summary",
+			  G_CALLBACK (app_state_changed), tile);
+	g_signal_connect (tile->app, "notify::key-colors",
+			  G_CALLBACK (app_state_changed), tile);
+	g_signal_connect (tile->app, "metadata-changed::GnomeSoftware::BackgroundTile-css",
+			  G_CALLBACK (app_background_tile_css_added), tile);
+	app_state_changed (tile->app, NULL, tile);
+
+	/* perhaps set custom css */
+	css = gs_app_get_metadata_item (app, "GnomeSoftware::BackgroundTile-css");
+	if (css != NULL)
+		gs_utils_widget_set_css (GTK_WIDGET (tile->image_box), css);
+	else
+		gs_background_tile_setup_colors_bg (tile);
+
+
+	if (gs_app_get_pixbuf (tile->app) != NULL)
+		gs_image_set_from_pixbuf (GTK_IMAGE (tile->image), gs_app_get_pixbuf (tile->app));
```

Comment:
Two calls to `gs_app_get_pixbuf()`? Maybe you could store the result in a variable and use that instead